src/query/service/src/sessions/query_ctx.rs

```diff
@@ -277,6 +279,16 @@ impl TableContext for QueryContext {
         Ok(())
     }
 
+    fn set_paritions_sha(&self, s: String) {
+        let mut sha = self.partitions_sha.write();
```

Comment:
What if there are multi tables? The `sha` could be override.

Reply:
Need to combine sha by order.

Reply:
Multi tables are not considered now. Will be considered in later PR.

Reply:
If it's not considered, we should have codes to forbid hitting the cache when querying on multi tables.